### PR TITLE
chore(maintenance): migrate logger utility to biome

### LINKED ISSUE #2798: Maintenance: add biome to project (logger)
### Summary

With #2791 being merged, we can now migrate the `packages/logger` workspace to use Biome.

As part of the migration I will fix any issue/inconsistency that arises. I'll do my best to minimize changes and leave the logic untouched.

### Why is this needed?

So that we can gradually replace `eslint`, `prettier`, and all their dependencies with a single one, `@biomejs/biome`.

### Which area does this relate to?

Other

### Solution

_No response_

### Acknowledgment

- [X] This request meets [Powertools for AWS Lambda (TypeScript) Tenets](https://docs.powertools.aws.dev/lambda/typescript/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Python](https://github.com/aws-powertools/powertools-lambda-python/), [Java](https://github.com/aws-powertools/powertools-lambda-java/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### Future readers

Please react with 👍 and your use case to help us understand customer demand.

### PR DESCRIPTION
**Pull request by daschaa**
### Changes

Adds biome to the logger package. Closes #2798 

**Issue number: 2798** 

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: packages/logger/src/types/Log.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import type { EnvironmentVariablesService } from '../config/EnvironmentVariablesService.js';
+import type { LogLevel as LogLevelList } from '../constants.js';
```
Comment (daschaa) [inline packages/logger/src/types/Log.ts:2]: The linter complained about LogLevel being declared twice, so I decided to give the import an alias in this file. Please let me know if you are happy with this or if it should be changed.

File: packages/logger/tests/unit/Logger.test.ts [hunk 2]
```diff
@@ -1,25 +1,25 @@
+import type { LambdaInterface } from '@aws-lambda-powertools/commons/types';
 /**
```
Comment (dreamorosi) [inline packages/logger/tests/unit/Logger.test.ts:2]: Could you please put the comment currently in L2 to L6 at the very top of the file? It's needed for Jest grouping, and it got moved by the Biome formatting.

**Review by dreamorosi — CHANGES_REQUESTED** [review]
Thank you for working on this!

I've just left one comment and then it's good to merge :)

File: packages/logger/tests/unit/Logger.test.ts (hunk 2 above)
Comment (daschaa) [reply packages/logger/tests/unit/Logger.test.ts]: Done :)

**Comment by dreamorosi** [comment]
@daschaa - Could you also take a look at the SonarCloud finding [here](https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=2813&id=aws-powertools_powertools-lambda-typescript&open=AZDfnaK1NKxyLq6_nzFi)?

I'll check your other two comments in the meanwhile - need to clone the repo to answer definitively.

**Comment by dreamorosi** [comment]
When you can, could you also pls leave a comment under the linked issue (& all others you're working or want to pick up) so that I can assign them to you? Thanks!

File: packages/logger/src/Logger.ts [hunk 3]
```diff
@@ -697,9 +692,11 @@ class Logger extends Utility implements LoggerInterface {
     const references = new WeakSet();
 
     return (key, value) => {
+      // biome-ignore lint/style/noParameterAssign:
```
[the remark this answers is not in the document]
Comment (dreamorosi) [reply packages/logger/src/Logger.ts:695]: I've pulled the branch and tested this, if we create a local var & use it throughout like this:

```ts
return (key, value) => {
  let replacedValue = value;
  if (this.#jsonReplacerFn) replacedValue = this.#jsonReplacerFn?.(key, replacedValue);

  if (replacedValue instanceof Error) {
    replacedValue = this.getLogFormatter().formatError(replacedValue);
  }
  if (typeof replacedValue === 'bigint') {
    return replacedValue.toString();
  }
  if (typeof replacedValue === 'object' && replacedValue !== null) {
    if (references.has(replacedValue)) {
      return;
    }
    references.add(replacedValue);
  }

  return replacedValue;
};
```

it's fine and the tests stay green - I'd say let's do something like this so we can remove the Biome suppression.

**Comment by daschaa** [comment]
@dreamorosi I'll check the sonarcloud issue and comment on the issues i'd like to pick :) Thank you for your support ❤️ Really awesome opportunity for me to learn about biome and to (hopefully) help you a bit out here :)

File: packages/logger/src/types/Log.ts (hunk 1 above)
Comment (dreamorosi) [reply packages/logger/src/types/Log.ts:2]: I think it's fine, good call!

**Comment by daschaa** [comment]
@dreamorosi Thanks for the review! 
- I have put the grouping comment to the top of each test file (btw I did not know about this feature in jest, really nice)
- I have removed the suppression of the variable override and introduced the local variable like you suggested
- I have fixed the SonarCloud finding by changing from a traditional for loop to a for...of loop

**Review by dreamorosi — APPROVED** [review]
Thank you for addressing all the feedback and for helping improve the project!

**Pull request merged**